include/ttmlir/Dialect/TT/IR/TTOps.td

```diff
@@ -6,5 +6,30 @@
 #define TTMLIR_TTMLIR_TTOPS_TD
 
 include "ttmlir/Dialect/TT/IR/TTOpsTypes.td"
+include "mlir/Interfaces/InferTypeOpInterface.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+include "mlir/IR/CommonTypeConstraints.td"
+
+def TT_GetTupleElementOp: TT_Op<"get_tuple_element", [Pure, DeclareOpInterfaceMethods<InferTypeOpInterface>]> {
```

Comment:
I'm wondering if this should go into TTNN dialect since it's really only used there.  What do you think?

Reply:
To me, it makes most sense if this op is in the same dialect as the Tuple itself. That way, any dialect consumer can include one dialect only and have "full" tuple functionality.

As to why TT over TTNN, I think it makes more sense as tuples are a generic thing and might be used by other dialects in the future. For this reason, I've tried to make the implementation generic, meaning the tuple can trivially be extended to hold any additional type. Moreover, in TT dialect we already have things like OOBVal, IteratorType, etc. which are also a kind of utility/auxiliary things.

Those were my thoughts. That said, if you still think TTNN is a better place, I'm not super opposed to moving both the tuple and the op to TTNN.

Reply:
Perhaps we should just use builtin tuple type?

https://mlir.llvm.org/docs/Dialects/Builtin/#tupletype

They do not however provide any operators for working with tuples, [rational](https://mlir.llvm.org/docs/Rationale/Rationale/#tuple-types), so we'd still need this op.

Yeah I think it's debatable, my original thinking was that this is specific to the TTNN dylib case, but it is purely utility so it doesn't hurt to have it in TT dialect, I'm ok with leaving it.

Reply:
Yeah, I've mentioned this in the summary of the PR 😊

Sounds good, I'll leave it as it is!